src/containers/blocks/UserBlock.js

```diff
@@ -0,0 +1,127 @@
+import React, {Component} from 'react';
+import {connect} from 'react-redux';
+import {bindActionCreators} from 'redux';
+import generateID from '../../functions/generateID';
+import {handleDragOver, handleDrop} from "../../functions/handleDragAndDrop";
+import {addNewFilter, toggleOptionsBox} from "../../actions";
+import DropContainer from "./DropBlock"
+
+class UserBlockContainer extends Component {
+
+  handleFilterDrop(ev, parentItem) {
+    const droppedItem = handleDrop(ev);
+    if(droppedItem) {
+      this.props.addNewFilter(ev, droppedItem, parentItem);
+    }
+  }
+
+  toggleOptionsBox(ev, test, ID, parentItem) {
+    ev.stopPropagation();
+    this.props.toggleOptionsBox(test, ID, parentItem);
+  }
+
+  generateListOfUserBlocks(type) {
+    if(this.props.test.length > 0) {
+      return this.props.test.map((test, index) => {
+        if(typeof test !== 'undefined') {   
+          if(test.dropTo.toLowerCase() === type) {
+            const elemClass = type + "-list-container";
+            let elemID = null;
+            if(test.type === "Source W3CHTML5") {
+              elemID = "source-comparators-" + index;
+            } else {
+              elemID = generateID(test) + "-" + index;
+            }
+            if(type === "comparators") {
+              return (
+              <div key={elemID} >
+                <div className={elemClass}>
+                  <div 
+                  className="block custom nested" 
+                  id={elemID} 
+                  onClick={(ev) => this.toggleOptionsBox(ev, test, elemID)}
+                  onDragOver={(ev) => handleDragOver(ev)}  
+                  onDrop={(ev) => this.handleFilterDrop(ev, test)}
+                  >{test.type}</div>
+                  {this.generateListOfFilters(test, index)}
+                </div>
+                <DropContainer dropTo={"comparators"} />
+              </div>
+              )
+            } else if(type === "modifiers") {
+              return (
```

Comment:
Both this and block below looks the same, only changed element is dropTo element that can be set conditially, I would see this logic changed somehow to avoid code duplication

Reply:
Done.